OpenRA.Game/Widgets/WidgetLoader.cs

```diff
@@ -48,6 +48,15 @@ public Widget LoadWidget(WidgetArgs args, Widget parent, MiniYamlNode node)
 			if (!args.ContainsKey("modRules"))
 				args = new WidgetArgs(args) { { "modRules", modData.DefaultRules } };
 
+			object parentLogic = null;
+			if (args.ContainsKey("logic"))
+			{
+				parentLogic = args["logic"];
+				args.Remove("logic");
+			}
+
+			args.Add("logic", node.Value.Nodes.FirstOrDefault(n => n.Key == "Logic"));
```

Comment:
Actually, we have exactly the same problem with the auto FieldLoader dictionary loading.  I think it would be best to fix and then use that here instead.